crates/cli/commands/src/common.rs

```diff
@@ -59,9 +59,12 @@ impl EnvironmentArgs {
         let db_path = data_dir.db();
         let sf_path = data_dir.static_files();
 
-        if access.is_read_write() {
-            reth_fs_util::create_dir_all(&db_path)?;
-            reth_fs_util::create_dir_all(&sf_path)?;
+        // ensure the provided datadir exist
+        if !db_path.is_dir() {
+            return Err(eyre::eyre!("Database path does not exist: {:?}", db_path));
+        }
+        if !sf_path.is_dir() {
+            return Err(eyre::eyre!("Static files path does not exist: {:?}", sf_path));
         }
```

Comment:
this isn't the right location to apply this check, because this is also used by other commands.

we should apply this here:

https://github.com/paradigmxyz/reth/blob/d41aac380bfebbbfd4208dff6f8bd0ceda00e9b1/crates/cli/commands/src/db/mod.rs#L66-L68

because all commands here are useless if the datadir does not exist

Reply:
you are right, updated the code